modules/light-clients/08-wasm/types/misbehaviour_handle_test.go

```diff
@@ -469,372 +481,128 @@ package types_test
 //				}
 //			}, false,
 //		},
-//	}
-//
-//	for _, tc := range testCases {
-//		suite.Run(tc.name, func() {
-//			suite.SetupWasmTendermint()
-//			path = ibctesting.NewPath(suite.chainA, suite.chainB)
-//
-//			err := path.EndpointA.CreateClient()
-//			suite.Require().NoError(err)
-//
-//			tc.malleate()
-//
-//			clientState := path.EndpointA.GetClientState()
-//			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
-//
-//			err = clientState.VerifyClientMessage(suite.chainA.GetContext(), suite.chainA.App.AppCodec(), clientStore, misbehaviour)
-//
-//			if tc.expPass {
-//				suite.Require().NoError(err)
-//			} else {
-//				suite.Require().Error(err)
-//			}
-//		})
-//	}
+// 	}
+
+// 	for _, tc := range testCases {
+// 		suite.Run(tc.name, func() {
+// 			// reset suite to create fresh application state
+// 			suite.SetupWasmWithMockVM()
+
+// 			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+// 			err := endpoint.CreateClient()
+// 			suite.Require().NoError(err)
+
+// 			tc.malleate()
+
+// 			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)
+// 			clientState := endpoint.GetClientState()
+// 			foundMisbehaviour := clientState.CheckForMisbehaviour(
+// 				suite.chainA.GetContext(),
+// 				suite.chainA.App.AppCodec(),
+// 				clientStore, // pass in clientID prefixed clientStore
+// 				clientMessage,
+// 			)
+
+// 			suite.Require().Equal(tc.foundMisbehaviour, foundMisbehaviour)
+// 		})
+// 	}
 // }
 
-/* func (suite *TypesTestSuite) TestCheckForMisbehaviourGrandpa() {
+func (suite *TypesTestSuite) TestCheckForMisbehaviour() {
 	var (
-		ok          bool
-		clientMsg   exported.ClientMessage
-		clientState exported.ClientState
+		foundMisbehaviour bool
+		panicErr          error
 	)
 
 	testCases := []struct {
-		name     string
-		malleate func()
-		expPass  bool
+		name                 string
+		malleate             func()
+		expFoundMisbehaviour bool
+		expPanic             bool
 	}{
 		{
-			"valid update no misbehaviour",
+			"no misbehaviour",
```

Comment:
```suggestion
			"success: no misbehaviour",
```
Someone mentioned that we were trying to use `success` and `failure` keywords in test names when I first joined. Can't remember who it was now

Reply:
i dont think if it makes sense to put success or failure in this case as the test case is for if `checkformisbehaviour` is working as expected right?

Reply:
You could put `failure` for panic case and `success` otherwise. This is basically what I did in my tests. But totally up to you